src/Illuminate/Support/Json.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Illuminate\Support;
+
+class Json
+{
+    /**
+     * Compile a PHP expression into a JavaScript object, array or single-quoted string.
+     *
+     * @param  mixed  $expression
+     * @param  int|null  $options
+     * @param  int  $depth
+     * @return string
+     */
+    public static function parse($expression, $options = null, $depth = 512)
+    {
+        if (is_object($expression) || is_array($expression)) {
+            $base64 = base64_encode(json_encode($expression, $options, $depth));
+
+            return "JSON.parse(atob('{$base64}'))";
```

Comment:
This line feels a bit of a hack to make it work with Alpine.js and only serves the cause of Alpine.js imho.

Can't it be made to work like a regular `json_encode` such that the output for
```js
<x-button x-data
   x-on:click="$dispatch( 'notify-errors', {{ js($errors->all()) }} )"
   type="submit"> Submit </x-button>
```

becomes simply an actual (correctly escaped if needed) javascript object like so:
```js
...
x-on:click="$dispatch( 'notify-errors', { someNumber: 5, someProp: [1,2,3] } )"
...
````

That would simplify this problem a lot since it directly aligns with the usage of `@json`.

Ofcourse it would be great if `@json` Blade directives (and others) could just be made to work but I guess thats a whole other issue.

Reply:
Specific to that `$errors` code snippet, the https://github.com/laravel/framework/blob/8.x/src/Illuminate/Support/ViewErrorBag.php class could be updated to meet PHP's JsonSerializable https://www.php.net/JsonSerializable interface.

```html
<!-- MessageBag@jsonSerialize() exists so this already works -->
@json($errors->getBag('default'))
```

```php
// new method on ViewErrorBag
public function jsonSerialize()
{
    return array_map(function (MessageBag $bag) {
        return $bag->jsonSerialize();
    }, $this->bags);
}
```

```html
@json($errors)
```

Personally, to serialize PHP for Vue JS components I added a custom Blade directive that invoked a simplified class derived from package "laracasts/utilities": https://github.com/laracasts/PHP-Vars-To-Js-Transformer/blob/1764a14305100ab2f2bea3e3197ff3400b4e66d7/src/Transformers/Transformer.php#L96.

Methods `jsonSerialize()` & `toJson()` + Blade `@json()` _should_ cover most uses. It's only a matter of if you don't own the class being serialized. e.g., from "laravel/framework" or a 3rd-party package.

Reply:
The problem is that `json_encode` returns double-quoted js. 
You can't use it in any html `data-foo="..."`. It's not just for Alpine.

Reply:
I have not looked into specifics for Alpine, but in Vue I can just use `<my-component :some-data='@json($data)' />`, note the single quotes. This works in every case, including for all kinds of quotes and characters inside your json strings. It helps that the default `@json` options default to `JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT`.